### ISSUE TITLE
Rasp. Stretch no USB device found

### ISSUE THREAD

**Issue by deve87** [karloygard/xcomfortd#7]
I got the ckoz0014 USB and get "Could not find/Open xComfort USB device"

dmesg shows

[  540.849685] usb 1-1.4: USB disconnect, device number 6
[  609.761297] usb 1-1.4: new full-speed USB device number 7 using dwc_otg
[  609.905527] usb 1-1.4: New USB device found, idVendor=188a, idProduct=1101, bcdDevice= 1.04
[  609.905541] usb 1-1.4: New USB device strings: Mfr=1, Product=2, SerialNumber=3
[  609.919388] hid-generic 0003:188A:1101.0002: hiddev96,hidraw0: USB HID v1.11 Device [EATON EATON Usb CI Stick    ] on usb-3f980000.usb-1.4/input0

**Comment by deve87** [comment]
Okay. Forgot to run as root ....

**Comment by karloygard** [comment]
Let me know how you get on with it; it's quirky to set up, but it's actually pretty reliable.

**Comment by deve87** [comment]
I will :-) Going to use node-red and Domoticz with this. So far I managed to turn ON / OFF switch actuator.

**Comment by karloygard** [comment]
Excellent! I only use it with home assistant myself. Be mindful of the fact that without the new actuators with extended status reporting, you can miss status changes if a user switches on/off lights with separate switches. Otherwise it works great.

Do you have the 00/13 stick to configure the network as well?

**Comment by deve87** [comment]
Yeah, I think MRF program saying something about statuslines when recalculateing signal / routing.

Yes, I use 13 USB stick for xComfort MRF program. I mainly use z-wave devices, but had some xComfort dimmers and switch actuators. I purchase the 14 USB stick when I found you'r code.

But I get MQTT disconnect after command is send. Do you know the reason?

**Comment by deve87** [comment]
Are you using a different system on Home Assistance? Sins you don't get disconnect issues?

**Comment by karloygard** [comment]
The MQTT protocol is handled completely by libmosquitto. It works well for me. You can try connecting it to a mosquitto broker and see if that works better?

**Comment by deve87** [comment]
I'm pretty new to MQTT, and I thought a mosquitto broker was the only way. 

After I run 'make' it gives me a runnable program xcomfortd witch connects to a (local) mosquitto broker.

./xcomfortd -v

**Comment by karloygard** [comment]
Mosquitto is just one of the available MQTT brokers. Just running `./xcomfortd -v` *should* work fine when the broker is running locally.

I am currently on vacation and away from the dev computer, so I can't really make any changes to the library. However, it would be interesting to see what libmosquitto logs in `mosquitto_log_callback_set()`. If you are proficient in C/C++, you can make the necessary changes yourself, in order to get a better idea what's going on.  Alternatively, I can make those changes when I get home from vacation, but that's a month away unfortunately.

**Comment by deve87** [comment]
I have no experience in C++. So if you have the time some day, would be great 🙂

Enjoy your vacation

**Comment by deve87** [comment]
I notised something after I programmed a xComfort switch to ckoz0014 via Eaton MRF.
When I started ./xcomfortd -v I get 

`MQTT Connected, Connection Accepted.`

When I send ex: 

`mosquitto_pub -h localhost -m 'true' -t 'xcomfort/1/set/switch'`

Switch actuator goes on and verbose shows:

`setting DP 1 to 1 (seq no 0, retry 0)`
`CKOZ-00/14 version numbers: RFV1.08, USBV1.04`

After that, I'm no longer abel to send any more commands.

But!

If I run ./xcomfortd -v and then press one of the buttons.

It's giving me:

`MQTT Connected, Connection Accepted.`
`CKOZ-00/14 version numbers: RFV1.08, USBV1.04`

And if I send an On / Off command after that and then, in between every command, pressing one of the switch buttons. It's actually works..

So it's like it waiting for a respons before it continues.

When I changes line 172:

`if (mosquitto_socket(mosq) != -1)`

to any other number than -1 (ex: 0)

Program works, but I still need to press switch (one time, on start) after running ./xcomfortd to make it work.
After that, I can set dimming values and turn on/off switches from Domoticz. . It will hang and disconnect from mosquitto after some minutes.
If run ex: true to a switch actuator and than true again. ./xcomfortd program will stop because it don't getting any respons back again because it's already on.
So if that happens, I can press switch button (before 60 sec keep alive), and program will continue before disconnecting.
(Because program wait for a response??)

Like in Issues post 2 (Messages are "delayed" or "off by one") I have the same problem. When I mention that I only needed to press switch one time, I actually need to press it twice becuse message are delayed.
